render-tests/runtime-styling/set-style-layer-change-source/style.json

```diff
@@ -0,0 +1,85 @@
+{
+  "version": 8,
+  "metadata": {
+    "ignored": {
+      "native": "https://github.com/mapbox/mapbox-gl-native/issues/2445"
+    },
+    "test": {
+      "width": 64,
+      "height": 64,
+      "operations": [
+        [
+          "setStyle",
+          {
+            "version": 8,
+            "transition": {
+              "duration": 0
+            },
+            "sources": {
+              "geojson": {
+                "type": "geojson",
+                "data": {
+                  "type": "Point",
+                  "coordinates": [
+                    0,
+                    0
+                  ]
+                }
+              },
+              "wrong-geojson": {
+                "type": "geojson",
+                "data": {
+                  "type": "Point",
+                  "coordinates": [
+                    5,
+                    5
+                  ]
+                }
+              }
+            },
+            "layers": [
+              {
+                "id": "circle",
+                "type": "circle",
+                "source": "geojson"
```

Comment:
Reason that `expected.png` looks right: the original `circle` layer points at `wrong-geojson`, whose coordinates are off-center.